src/Microsoft.AspNetCore.Routing.Abstractions/EndpointMetadataCollection.cs

```diff
@@ -24,15 +39,34 @@ public EndpointMetadataCollection(IEnumerable<object> items)
             _items = items.ToArray();
         }
 
+        /// <summary>
+        /// Creates a new <see cref="EndpointMetadataCollection"/>.
+        /// </summary>
+        /// <param name="items">The metadata items.</param>
         public EndpointMetadataCollection(params object[] items)
             : this((IEnumerable<object>)items)
         {
         }
 
+        /// <summary>
+        /// Gets the item at <paramref name="index"/>.
+        /// </summary>
+        /// <param name="index">The index of the item to retrieve.</param>
+        /// <returns>The item at <paramref name="index"/>.</returns>
         public object this[int index] => _items[index];
 
+        /// <summary>
+        /// Gets the count of metadata items.
+        /// </summary>
         public int Count => _items.Length;
 
+        /// <summary>
+        /// Gets the most significant metadata item of type <typeparamref name="T"/>.
```

Comment:
Regarding: `most significant metadata`, I feel can example or bit more explanation can help

Reply:
It's explained more on the class heading